olive/systems/ort_inference/ort_inference_system.py

```diff
@@ -0,0 +1,271 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import collections
+import json
+import logging
+import tempfile
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Callable, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+from torch.utils.data import Dataset
+
+from olive.common.utils import run_subprocess
+from olive.evaluator.metric import get_latency_config_from_metric
+from olive.evaluator.olive_evaluator import OliveEvaluator, OliveModelOutput, OnnxEvaluatorMixin
+from olive.hardware import Device
+from olive.systems.common import SystemType
+from olive.systems.olive_system import OliveSystem
+from olive.systems.system_config import ORTInferenceTargetUserConfig
+from olive.systems.utils import create_new_environ, run_available_providers_runner
+
+if TYPE_CHECKING:
+    from olive.evaluator.metric import Metric, MetricResult
+    from olive.hardware.accelerator import AcceleratorSpec
+    from olive.model import ModelConfig, ONNXModelHandler
+    from olive.passes.olive_pass import Pass
+
+logger = logging.getLogger(__name__)
+
+
+class ORTInferenceSystem(OliveSystem):
+    system_type = SystemType.ORTInference
+
+    def __init__(
+        self,
+        python_environment_path: Union[Path, str] = None,
+        environment_variables: Dict[str, str] = None,
+        prepend_to_path: List[str] = None,
+        accelerators: List[str] = None,
+        hf_token: bool = None,
+    ):
+        super().__init__(accelerators=accelerators, olive_managed_env=False)
+        self.config = ORTInferenceTargetUserConfig(
+            python_environment_path=python_environment_path,
+            environment_variables=environment_variables,
+            prepend_to_path=prepend_to_path,
+            accelerators=accelerators,
+        )
+        self.environ = create_new_environ(
+            python_environment_path=self.config.python_environment_path,
+            environment_variables=self.config.environment_variables,
+            prepend_to_path=self.config.prepend_to_path,
+        )
+
+        # available eps. This will be populated the first time self.get_supported_execution_providers() is called.
+        # used for caching the available eps
+        self.available_eps = None
+
+    def run_pass(
+        self,
+        the_pass: "Pass",
+        model_config: "ModelConfig",
+        data_root: str,
+        output_model_path: str,
+        point: Optional[Dict[str, Any]] = None,
+    ) -> "ModelConfig":
+        """Run the pass on the model at a specific point in the search space."""
+        logger.warning("ORTInferenceSystem does not support running passes.")
+        raise NotImplementedError
```

Comment:
does it need run_pass?

Reply:
Same as above, Olive cannot be installed in this environment. So we cannot run pass unless we fallback on the default local system running the engine.

Since we only expect to use this system for qnn ep, we could ignore this system for the 'run_on_target' feature that we discussed before? 
We don't expect to use perf tuning or inc for this ep target.